Protocol.md

```diff
@@ -279,6 +279,50 @@ The recipient:
 
   These script pub key forms include only standard forms accepted by the wider set of deployed Bitcoin clients in the network, which increase the chances of successful propagation to miners.
 
+## Second Layer Interaction
+
+Second Layer Interaction consist of second layer "features" which parties can opt-in to using. These messages are not needed to facilitate construction of a DLCs.
+
+### The `close_dlc` Message
+
+This message contains information about mutual close, which allows the counterparty
+to broadcast a mutual closing transaction.
+
+1. type: ? (`close_dlc_v0`)
+2. data:
+   * [`32*byte`:`contract_id`]
+   * [`signature`:`close_signature`]
+   * [`u64`:`offerPayoutSatoshis`]
+   * [`u64`:`acceptPayoutSatoshis`]
+   * [`u64`:`fund_input_serial_id`]
+   * [`u16`:`num_funding_inputs`]
+   * [`num_funding_inputs*funding_input`:`funding_inputs`]
+
+
+`payout_spk` and `payout_serial_id` from `offer_dlc` as well as `payout_spk` and `payout_serial_id` from `accept_dlc` should be used for constructing the close transaction
+
+`fund_input_serial_id` is a randomly chosen number which uniquely identifies the funding input of the initiating party.
+Inputs in the closing transaction will be sorted by `fund_input_serial_id` and `input_serial_id` in `funding_inputs`.
+
+`funding_inputs` are extra inputs to mutual close to avoid free option, and in the future revocation mechanism will be introduced
+
+#### Requirements
+
+The sender MUST:
+
+  - set `contract_id` by exclusive-OR of the `funding_txid`, the `funding_output_index` and the `temporary_contract_id` from the `offer_dlc` and `accept_dlc` messages.
+  - set `close_signature` to the valid signature, using its `funding_pubkey` for the close transaction, as defined in the [transaction specification](Transactions.md#close-transaction).
+
+The recipient:
+
+  - if any input in `funding_inputs` is not a BIP141 (Segregated Witness) input.
+    - MUST reject the contract.
+  - if the `close_signature` is incorrect:
+    - MUST reject the contract.
+  - MUST NOT broadcast the closing transaction before receipt of a valid `close_dlc`.
+  - on receipt of a valid `close_dlc`:
+    - SHOULD broadcast the closing transaction.
```

Comment:
Is it though? What if the counter party doesn't want to close? Or are you considering a setting where there is implicit trust between the parties that they will close when requested? In that case it would be better to specify it, but maybe better to be general and not required that the transaction be broadcast.
That actually make me think that in practice it might be better to have first a "propose close" message to check if the counter party is willing to close or not. But I guess that's maybe to complex for now.

Reply:
In the particular setting I had in mind, there would probably be a level of trust between the parties i.e. market maker and user. However you're right, this probably doesn't belong in the protocol. Additionally, even if there is trust, there's no way for the initiator to know if `offerPayoutSatoshis` and `acceptPayoutSatoshis` are set to an amount that the counterparty will accept since the market can change quickly. 

I agree that a "propose close" message would be better to check if the other party is open to closing, but maybe out of scope for this PR. On a separate note, there should probably be a "propose dlc offer" message so that a party can check if a counter party is willing to enter into a dlc with them without revealing inputs.

Reply:
Yeah it's tricky... I don't have a really good solution for now.

Reply:
Gonna resolve this. My thinking now is that a "propose close" of "propose dlc offer" are things that should happen outside of the core protocol (at least for now).